starter-cli/src/main/java/io/micronaut/starter/cli/feature/messaging/jms/CreateJmsConsumer.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.cli.feature.messaging.jms;
+
+import com.fizzed.rocker.RockerModel;
+import io.micronaut.context.annotation.Parameter;
+import io.micronaut.context.annotation.Prototype;
+import io.micronaut.core.annotation.ReflectiveAccess;
+import io.micronaut.core.util.functional.ThrowingSupplier;
+import io.micronaut.starter.application.Project;
+import io.micronaut.starter.cli.CodeGenConfig;
+import io.micronaut.starter.cli.command.CodeGenCommand;
+import io.micronaut.starter.cli.feature.messaging.jms.template.listener.groovyListener;
+import io.micronaut.starter.cli.feature.messaging.jms.template.listener.javaListener;
+import io.micronaut.starter.cli.feature.messaging.jms.template.listener.kotlinListener;
+import io.micronaut.starter.io.ConsoleOutput;
+import io.micronaut.starter.io.OutputHandler;
+import io.micronaut.starter.options.Language;
+import io.micronaut.starter.template.RenderResult;
+import io.micronaut.starter.template.RockerTemplate;
+import io.micronaut.starter.template.TemplateRenderer;
+import picocli.CommandLine;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+@CommandLine.Command(name = "create-jms-consumer", description = "Creates a consumer class for JMS")
+@Prototype
+public class CreateJmsConsumer extends CodeGenCommand {
+
+    @ReflectiveAccess
+    @CommandLine.Parameters(paramLabel = "CONSUMER", description = "The name of the consumer to create")
+    String consumerName;
+
+    @Inject
+    public CreateJmsConsumer(@Parameter CodeGenConfig config) {
+        super(config);
+    }
+
+    public CreateJmsConsumer(CodeGenConfig config,
+                             ThrowingSupplier<OutputHandler, IOException> outputHandlerSupplier,
+                             ConsoleOutput consoleOutput) {
+        super(config, outputHandlerSupplier, consoleOutput);
+    }
+
+    @Override
+    public boolean applies() {
+        return config.getFeatures().contains("jms-activemq-artemis")
```

Comment:
one small tweak - the 3 feature classes use a constant for the name, e.g. https://github.com/micronaut-projects/micronaut-starter/blob/2.3.x/starter-core/src/main/java/io/micronaut/starter/feature/messaging/jms/ActiveMqArtemis.java#L26, so using those instead of hard-coded values would be a bit safer

Reply:
done…thanks for the review